### PR TITLE
impl Machine::{new, run}

### PR DESCRIPTION
**Pull request by memoryleak47**

### CODE REVIEW

File: mem/basic.md [hunk 1]
```diff
@@ -54,6 +54,14 @@ impl Memory for BasicMemory {
 }
 ```
 
+```rust
+impl Memory for BasicMemory {
+    fn new() -> Self {
+        Self { allocations: List::new() }
+    }
+}
+```
```
Comment (memoryleak47) [inline mem/basic.md:63]: Do you prefer `Memory: Default` and `#[derive(Default)]` to make this a bit more concise?

Comment (RalfJung) [reply mem/basic.md:63]: I think I prefer `new`, so this looks good :)

File: lang/machine.md [hunk 2]
```diff
@@ -43,6 +43,44 @@ struct StackFrame<M: Memory> {
...
+            addr: Int::ZERO,
+            provenance: None
+        };
+
+        // setup the initial stack frame.
+        let init_frame = StackFrame {
+            func: start_fn,
+            locals: Map::new(),
+            // The initial function has no `caller_ret_place`.
+            caller_ret_place: null_ptr,
```
Comment (RalfJung) [inline lang/machine.md:63]:
```suggestion
            // The initial function has no caller and hence no `caller_ret_place`.
            caller_ret_place: null_ptr,
```

Comment (memoryleak47) [reply lang/machine.md]: Yep, that sounds better. It's squashed in!

**Comment by memoryleak47** [comment]
@RalfJung ready for review :)

**Comment by RalfJung** [comment]
I have adjusted this a bit more in 7e50767... hopefully it still builds. ;)

**Comment by memoryleak47** [comment]
> hopefully it still builds. ;)

almost ^^

**Comment by RalfJung** [comment]
Damn... we need that CI.  ;)

**Pull request merged**